uiServices/src/services.go

```diff
@@ -96,5 +94,141 @@ func validateJsonWithSchema(config []byte) bool {
         }
     }
     return true
+}
+
+func getConfigs(rw http.ResponseWriter, req *http.Request){
+
+    configPathDir := req.Header.Get("configPathDir")
+    configName := chi.URLParam(req, "configName")
+
+    if !strings.HasSuffix(configPathDir, "/"){
+        configPathDir = configPathDir + "/"
+    }
+
+    file, err := os.Open(fmt.Sprintf("%s%s.xml", configPathDir, configName))
+    if len(configPathDir) <= 1{
+        rw.WriteHeader(http.StatusBadRequest)
+        logrus.Error("No Header Path Found")
+        return
+    }
+    if err != nil {
+        logrus.Error("Configuration Name Not Found: "+configPathDir + configName)
+        rw.WriteHeader(http.StatusNotFound)
+        return
+    }
+    defer file.Close()
+
+    if err != nil {
+        logrus.Error("Configuration Name Not Found: "+configPathDir + configName)
+        rw.WriteHeader(http.StatusNotFound)
+        return
+    }
+
+    var config perfTestUtils.Config
+
+    byteValue, err := ioutil.ReadAll(file)
+    if err != nil{
+        rw.WriteHeader(http.StatusInternalServerError)
+        logrus.Error("Cannot Read File")
+        return
+    }
+
+    err = xml.Unmarshal(byteValue, &config)
+    if err != nil{
+        rw.WriteHeader(http.StatusInternalServerError)
+        logrus.Error("Cannot Unmarshall")
+        return
+    }
+ 
+    configJson, err := json.MarshalIndent(config,"","")
+    if err != nil {
+        rw.WriteHeader(http.StatusInternalServerError)
+        logrus.Error("Cannot Marshall")
+        return
+    }
+
+    rw.WriteHeader(http.StatusOK)
+    rw.Write(configJson)
+    logrus.Println(string(configJson))
 
 }
+
+func putConfigs(rw http.ResponseWriter, req *http.Request) {
+    path := req.Header.Get("configPathDir")
```

Comment:
enhancement:
create a function "getConfigPathHeader" which returns the header string value and an error if not found. Then you can move a lot of the logic(below) into a separate function that is shared between getConfigs and PutConfigs.

    path := req.Header.Get("configPathDir")
      if !strings.HasSuffix(path, "/") {
             path = path + "/"
         }
 
     if len(path) <= 1 {
         logrus.Error("No Path Entered")
         rw.WriteHeader(http.StatusBadRequest)
         return
     }